### DISCUSSION TITLE
How to change languages based on the User object without locale prefixes

### DISCUSSION THREAD

**Discussion by christinasirb** [amannn/next-intl#778] [Ideas]
Hello!

I am using next-intl for my app and I need a way to change between languages, without adding /de or /en to the path. I need to set the locale based on the authenticated user's preferences. How can I change the locale?

Thanks!

**Comment by markomitranic** [comment]
That is a very interesting use case!

I'm not sure about the whole routing part, someone else can answer that. I guess you would simply disable prefixes and disable rewrites?

As for getting the information from a user object, I've done that in the past and it is a very common thing - users can choose which language they prefer, which has higher seniority than cookies, geoip or header. I believe that in this case you'd ["compose middleware"](https://next-intl-docs.vercel.app/docs/routing/middleware#composing-other-middlewares) to override the locale that the middleware has set. Alternatively, (and perhaps simpler) you can simply forcibly set a cookie based on the user's preferences :D

**Comment by J4v4Scr1pt** [comment]
I'm also trying create a nice solution on this matter. Some good docs on this would be super nice.

**Comment by amannn** [comment]
This question just came up today again, see my answer here: https://github.com/amannn/next-intl/discussions/892#discussioncomment-8607185

**Reply by amannn** [reply]
I've opened https://github.com/amannn/next-intl/issues/960 to find a better solution for this use case.
